test/grad_test.jl

```diff
@@ -0,0 +1,71 @@
+using Test
+using Printf
+using LinearAlgebra
```

Comment:
These 'using' statements are not needed here?

Reply:
Oh yeah, good point. I like them because it makes this file work as alone instead of depending on external files, but I do see that this is not standard Julia convention, so I'll get rid of them.